Migration of the data source aci_system into plugin framework (DCNE-163) (#1279)
CiscoDevNet/terraform-provider-aci: docs/data-sources/system.md

* `pod_id` (podId) - (string) The pod identifier.

### Read-Only ###

* `id` - (string) The distinguished name (DN) of the system object.
* `address` (address) - (string) The IP address of the system.
* `bootstrap_state` (bootstrapState) - (string) Bootstrap state of this system.
* `cluster_time_diff` (clusterTimeDiff) - (string) Difference in cluster time from local time for this system.
* `control_plane_mtu` (controlPlaneMTU) - (string) MTU for control plane (SUP-originated) packets.
* `current_time` (currentTime) - (string) The current time on this system.
* `enforce_subnet_check` (enforceSubnetCheck) - (string) Enforce subnet check on all VRFs.
* `external_tep_address` (etepAddr) - (string) The external TEP IP address of this pod.
* `fabric_domain` (fabricDomain) - (string) Fabric domain of this node.
* `fabric_id` (fabricId) - (string) The latest system health score.
* `fabric_mac` (fabricMAC) - (string) The MAC address of the fabric.
* `inband_management_address` (inbMgmtAddr) - (string) The In-band management IPv4 address.
* `inband_management_address_ipv6` (inbMgmtAddr6) - (string) The In-band management IPv6 address.

> * `inband_management_address_mask_ipv6` (inbMgmtAddr6Mask) - (string) The In-band management IPv6 address Mask.

Reviewer: ```suggestion
* `inband_management_address_mask_ipv6` (inbMgmtAddr6Mask) - (string) The In-band management IPv6 address subnet mask.
```

> * `inband_management_address_mask` (inbMgmtAddrMask) - (string) The In-band management IP address Mask.

Reviewer: ```suggestion
* `inband_management_address_mask` (inbMgmtAddrMask) - (string) The In-band management IPv4 address subnet mask.
```

> * `inband_management_gateway` (inbMgmtGateway) - (string) The In-band management IP Gateway.

Reviewer: ```suggestion
* `inband_management_gateway` (inbMgmtGateway) - (string) The In-band management IPv4 gateway address.
```

> * `inband_management_gateway_ipv6` (inbMgmtGateway6) - (string) The In-band management IPv6 Gateway.

Reviewer: ```suggestion
* `inband_management_gateway_ipv6` (inbMgmtGateway6) - (string) The In-band management IPv6 gateway address.
```

* `last_reboot_time` (lastRebootTime) - (string) The last reboot time for this system.
* `last_reset_reason` (lastResetReason) - (string) The last reset reason for this system.
* `mod_ts` (modTs) - (string) The time when this object was last modified.
* `mode` (mode) - (string) Specifies if this system is configured in standalone mode or HA pair.
* `monitoring_policy_dn` (monPolDn) - (string) The monitoring policy attached to this observable object.
* `name` (name) - (string) The name of the system object.
* `name_alias` (nameAlias) - (string) The name alias of the system object.
* `node_type` (nodeType) - (string) The role of this system.
* `out_of_band_management_address` (oobMgmtAddr) - (string) The Out-of-band management IPv4 address.
* `out_of_band_management_address_ipv6` (oobMgmtAddr6) - (string) The Out-of-band management IPv6 address.

> * `out_of_band_management_address_mask_ipv6` (oobMgmtAddr6Mask) - (string) The Out-of-band management IPv6 address Mask.

Reviewer: ```suggestion
* `out_of_band_management_address_mask_ipv6` (oobMgmtAddr6Mask) - (string) The Out-of-band management IPv6 address subnet mask.
```

> * `out_of_band_management_address_mask` (oobMgmtAddrMask) - (string) The Out-of-band management IP address Mask.

Reviewer: ```suggestion
* `out_of_band_management_address_mask` (oobMgmtAddrMask) - (string) The Out-of-band management IPv4 address subnet mask.
```

> * `out_of_band_management_gateway` (oobMgmtGateway) - (string) The Out-of-band management IP Gateway.

Reviewer: ```suggestion
* `out_of_band_management_gateway` (oobMgmtGateway) - (string) The Out-of-band management IPv4 gateway address.
```

> * `out_of_band_management_gateway_ipv6` (oobMgmtGateway6) - (string) The Out-of-band management IPv6 Gateway.

Reviewer: ```suggestion
* `out_of_band_management_gateway_ipv6` (oobMgmtGateway6) - (string) The Out-of-band management IPv6 gateway address.
```

* `remote_network_id` (remoteNetworkId) - (string) The remote network ID.
* `remote_node` (remoteNode) - (string) The remote system.
* `remote_leaf_auto_mode` (rlAutoMode) - (string) The remote leaf auto mode.
* `remote_leaf_group_id` (rlGroupId) - (string) The remote leaf site identifier.

> * `remote_leaf_operational_pod_id` (rlOperPodId) - (string) The operational pod idenitfier for RL pod redundancy.

Reviewer: ```suggestion
* `remote_leaf_operational_pod_id` (rlOperPodId) - (string) The operational pod identifier for RL pod redundancy.
```

> * `remote_leaf_routable_mode` (rlRoutableMode) - (string) Indicates whether the remote leaf is routable.

Reviewer: ```suggestion
* `remote_leaf_routable_mode` (rlRoutableMode) - (string) Indicates whether the remote leaf is using a routable TEP IP address.
```

* `remote_leaf_direct_mode` (rldirectMode) - (string) The remote leaf direct mode.
* `role` (role) - (string) The system role type.
* `serial` (serial) - (string) The serial number of the system.
* `server_type` (serverType) - (string) DHCP server type.
* `site_id` (siteId) - (string) The site identifier.
* `state` (state) - (string) The operational state of this system.

> * `system_up_time` (systemUpTime) - (string) The time (in seconds) since the system was booted.

Reviewer: ```suggestion
* `system_uptime` (systemUpTime) - (string) The time (in seconds) since the system was booted.
```

* `tep_pool` (tepPool) - (string) The pool of TEP IP addresses allocated for this system.

> * `unicast_xr_ep_learn_disable` (unicastXrEpLearnDisable) - (string) Indicates whether the learning of unicast endpoints for external routing is disabled.

Reviewer: ```suggestion
* `unicast_xr_endpoint_learn_disable` (unicastXrEpLearnDisable) - (string) Indicates whether the learning of unicast endpoints for external routing is disabled.
```

  Data source for ACI System
---

> # aci_system #

Author: Initially generated it and made a few changes to make it work. We decided to keep this independent from auto generation like you suggested.

Data source for ACI System

## API Information ##

* Class: [topSystem](https://pubhub.devnetcloud.com/media/model-doc-latest/docs/app/index.html#/objects/topSystem/overview)

* Supported in ACI versions: 1.0(1e) and later.

* Distinguished Name Formats:
  - `sys`
  - `topology/pod-{id}/node-{id}/sys`

## GUI Information ##

* Location: `Generic`

## Example Usage ##

```hcl

data "aci_system" "example" {
  pod_id    = "1"
  system_id = "1"
}

```

## Schema ##

### Required ###

* `system_id` (id) - (string) The identifier of the System object.

> * `pod_id` (podId) - (string) POD Idenitfier.

Reviewer 2: Think pod should be lower case? Also should we use identifier or ID?
```suggestion
* `pod_id` (podId) - (string) The pod identifier.
```

### Read-Only ###

* `id` - (string) The distinguished name (DN) of the System object.
* `address` (address) - (string) The IP address of the system.
* `bootstrap_state` (bootstrapState) - (string) Bootstrap state of this system.
* `cluster_time_diff` (clusterTimeDiff) - (string) Difference in cluster time from local time for this system.
* `control_plane_mtu` (controlPlaneMTU) - (string) MTU for control plane (SUP-originated) packets.
* `current_time` (currentTime) - (string) The current time on this system.
* `enforce_subnet_check` (enforceSubnetCheck) - (string) Enforce subnet check on all VRFs.

> * `etep_addr` (etepAddr) - (string) The External TEP IP address of this PoD.

Reviewer 2: ```suggestion
* `external_tep_address` (etepAddr) - (string) The External TEP IP address of this PoD.
```

* `fabric_domain` (fabricDomain) - (string) Fabric domain of this node.
* `fabric_id` (fabricId) - (string) The latest system health score. Use the navigation bar at the top right of the table to select which health level to view.
* `fabric_mac` (fabricMAC) - (string) 
* `inb_mgmt_addr` (inbMgmtAddr) - (string) The in-band management IPv4 address.
* `inb_mgmt_addr6` (inbMgmtAddr6) - (string) In-band management IPv6 address.

> * `inb_mgmt_addr6_mask` (inbMgmtAddr6Mask) - (string) In-band management IPv6 address Mask.

Reviewer 2: Some sentences start with The and other not, perhaps from consistency make it similar?

> * `inb_mgmt_addr` (inbMgmtAddr) - (string) The in-band management IPv4 address.
> * `inb_mgmt_addr6` (inbMgmtAddr6) - (string) In-band management IPv6 address.
> * `inb_mgmt_addr6_mask` (inbMgmtAddr6Mask) - (string) In-band management IPv6 address Mask.
> * `inb_mgmt_addr_mask` (inbMgmtAddrMask) - (string) In-band management IP address Mask.
> * `inb_mgmt_gateway` (inbMgmtGateway) - (string) In-band management IP Gateway.
> * `inb_mgmt_gateway6` (inbMgmtGateway6) - (string) In-band management IPv6 Gateway.

Reviewer 2: ```suggestion
* `inband_management_address` (inbMgmtAddr) - (string) The in-band management IPv4 address.
* `inband_management_address_ipv6` (inbMgmtAddr6) - (string) In-band management IPv6 address.
* `inband_management_address_mask_ipv6` (inbMgmtAddr6Mask) - (string) In-band management IPv6 address Mask.
* `inband_management_address_mask` (inbMgmtAddrMask) - (string) In-band management IP address Mask.
* `inband_management_gateway` (inbMgmtGateway) - (string) In-band management IP Gateway.
* `inband_management_gateway_ipv6` (inbMgmtGateway6) - (string) In-band management IPv6 Gateway.
```

> * `last_reboot_time` (lastRebootTime) - (string) LastRebootTime for this system.
> * `last_reset_reason` (lastResetReason) - (string) LastResetReason for this system.

Reviewer 2: Do we name it system or should we use APIC?
```suggestion
* `last_reboot_time` (lastRebootTime) - (string) The last reboot time for this system.
* `last_reset_reason` (lastResetReason) - (string) The last reset reason for this system.
```

> * `lc_own` (lcOwn) - (string) A value that indicates how this object was created. For internal use only.

Author: removed lc_own..

> * `lc_own` (lcOwn) - (string) A value that indicates how this object was created. For internal use only.

Author: I removed a few others as well but missed this one

* `mod_ts` (modTs) - (string) The time when this object was last modified.
* `mode` (mode) - (string) Specifies if this system is configured in standalone mode or HA pair.

> * `mon_pol_dn` (monPolDn) - (string) The monitoring policy attached to this observable object.

Reviewer 2: ```suggestion
* `monitoring_policy_dn` (monPolDn) - (string) The monitoring policy attached to this observable object.
```

* `name` (name) - (string) The name of the System object.
* `name_alias` (nameAlias) - (string) The name alias of the System object.
* `node_type` (nodeType) - (string) Role of this system.
* `oob_mgmt_addr` (oobMgmtAddr) - (string) The out-of-band management IPv4 address.
* `oob_mgmt_addr6` (oobMgmtAddr6) - (string) Out-of-band management IPv6 address.
* `oob_mgmt_addr6_mask` (oobMgmtAddr6Mask) - (string) Out-of-band management IPv6 address Mask.
* `oob_mgmt_addr_mask` (oobMgmtAddrMask) - (string) Out-of-band management IP address Mask.
* `oob_mgmt_gateway` (oobMgmtGateway) - (string) Out-of-band management IP Gateway.

> * `oob_mgmt_gateway6` (oobMgmtGateway6) - (string) Out-of-band management IPv6 Gateway.

Reviewer 2: same as inband

* `remote_network_id` (remoteNetworkId) - (string) Remote Network ID.
* `remote_node` (remoteNode) - (string) Remote system.

> * `rl_auto_mode` (rlAutoMode) - (string) Remote Leaf Auto Mode.
> * `rl_group_id` (rlGroupId) - (string) Remote Leaf Site Identifier.
> * `rl_oper_pod_id` (rlOperPodId) - (string) Operational POD Idenitfier for RL Pod Redundancy.
> * `rl_routable_mode` (rlRoutableMode) - (string) Is RemoteLeaf Routable.
> * `rldirect_mode` (rldirectMode) - (string) Remote Leaf Direct Mode.

Reviewer 2: also adjust descriptions
```suggestion
* `remote_leaf_auto_mode` (rlAutoMode) - (string) Remote Leaf Auto Mode.
* `remote_leaf_group_id` (rlGroupId) - (string) Remote Leaf Site Identifier.
* `remote_leaf_operational_pod_id` (rlOperPodId) - (string) Operational POD Idenitfier for RL Pod Redundancy.
* `remote_leaf_routable_mode` (rlRoutableMode) - (string) Is RemoteLeaf Routable.
* `remote_leaf_direct_mode` (rldirectMode) - (string) Remote Leaf Direct Mode.
```

* `role` (role) - (string) The system role type.
* `serial` (serial) - (string) Serial Number of the system.
* `server_type` (serverType) - (string) DhcpServerType.
* `site_id` (siteId) - (string) Site Identifier.
* `state` (state) - (string) Operational state of this system.
* `system_up_time` (systemUpTime) - (string) The time (in seconds) since the system was booted.
* `tep_pool` (tepPool) - (string) TepPool for this system.

> * `unicast_xr_ep_learn_disable` (unicastXrEpLearnDisable) - (string) Disable xrLeanrs.

Reviewer 2: what does xr stand for in xrlearn?

Author: I looked it up and it is external routing..updated it accordingly

* `version` (version) - (string) version for this system.

> * `virtual_mode` (virtualMode) - (string) Virtual mode of system.

Reviewer 2: ```suggestion
* `virtual_mode` (virtualMode) - (string) The virtual mode of this system.
```
